contribs/gmf/src/directives/map.js

```diff
@@ -31,45 +37,81 @@ goog.require('ol.Map');
 gmf.mapDirective = function() {
   return {
     scope: {
-      'getMapFn': '&gmfMapMap'
+      'map': '<gmfMapMap',
+      'manageResize': '<gmfMapManageResize',
+      'resizeTransition': '<gmfMapResizeTransition'
     },
     controller: 'GmfMapController as ctrl',
-    template: '<div ngeo-map="ctrl.map"></div>'
+    bindToController: true,
+    templateUrl: `${gmf.baseTemplateUrl}/map.html`
   };
 };
 
 gmf.module.directive('gmfMap', gmf.mapDirective);
 
 
 /**
- * @param {angular.Scope} $scope The directive's scope.
- * @param {ngeo.FeatureOverlayMgr} ngeoFeatureOverlayMgr The ngeo feature
- * @param {gmf.Permalink} gmfPermalink The gmf permalink service.
- * @param {gmf.Snapping} gmfSnapping The gmf snapping service.
+ * @param {!ngeo.FeatureOverlayMgr} ngeoFeatureOverlayMgr The ngeo feature
+ * @param {!gmf.Permalink} gmfPermalink The gmf permalink service.
+ * @param {!gmf.Snapping} gmfSnapping The gmf snapping service.
  * @constructor
  * @private
  * @ngInject
  * @ngdoc controller
  * @ngname GmfMapController
  */
-gmf.MapController = function($scope, ngeoFeatureOverlayMgr, gmfPermalink,
-    gmfSnapping) {
+gmf.MapController = function(ngeoFeatureOverlayMgr, gmfPermalink, gmfSnapping) {
 
-  const map = $scope['getMapFn']();
-  goog.asserts.assertInstanceof(map, ol.Map);
+  // Scope properties
 
   /**
    * @type {!ol.Map}
    * @export
    */
-  this.map = map;
+  this.map;
+
+  /**
+   * @type {boolean|undefined}
+   * @export
+   */
+  this.manageResize;
+
+  /**
+   * @type {boolean|undefined}
+   * @export
+   */
+  this.resizeTransition;
 
-  ngeoFeatureOverlayMgr.init(this.map);
 
-  gmfPermalink.setMap(this.map);
+  // Injected properties
+
+  /**
+   * @type {!ngeo.FeatureOverlayMgr}
+   * @private
+   */
+  this.ngeoFeatureOverlayMgr_ = ngeoFeatureOverlayMgr;
 
-  gmfSnapping.setMap(this.map);
+  /**
+   * @type {!gmf.Permalink}
+   * @private
+   */
+  this.gmfPermalink_ = gmfPermalink;
 
+  /**
+   * @type {!gmf.Snapping}
+   * @private
+   */
+  this.gmfSnapping_ = gmfSnapping;
 };
 
 gmf.module.controller('GmfMapController', gmf.MapController);
+
+
+/**
+ * Called on initialization of the controller.
+ */
+gmf.MapController.prototype.$onInit = function() {
```

Comment:
There are other directives that use controllers with this `$onInit` method.  You're right, it's odd since it seems to be documented in the components only.  But, it's a controller's method.

Should we leave this as it is?